pkg/services/serviceaccounts/manager/service.go

```diff
@@ -39,6 +39,28 @@ func ProvideServiceAccountsService(
 	}
 	serviceaccountsAPI := api.NewServiceAccountsAPI(s, ac, routeRegister)
 	serviceaccountsAPI.RegisterAPIEndpoints(cfg)
+
+	basicKeys := store.GetNonServiceAccountAPIKeys(context.Background())
+	if len(basicKeys) > 0 {
+		s.log.Info("Launching background thread to upgrade API keys to service accounts", "numberKeys", len(basicKeys))
+		go func() {
+			for _, key := range basicKeys {
+				sa, err := store.CreateServiceAccountForApikey(context.Background(), key.OrgId, key.Name, key.Role)
+				if err != nil {
+					s.log.Error("Failed to create service account for API key", "err", err, "keyId", key.Id)
+					continue
+				}
```

Comment:
hmm 🤔 should we capture these or so you think?

basically do we want to continue in case there is one or more accounts that failed.

Reply:
The user can't do anything to fix it , so we should continue.